remotes/docker/resolver.go

```diff
@@ -283,8 +283,12 @@ func (r *dockerResolver) Resolve(ctx context.Context, ref string) (string, ocisp
 			log.G(ctx).Debug("resolving")
 			resp, err := req.doWithRetries(ctx, nil)
 			if err != nil {
+				if lastErr == nil {
+					lastErr = err
+				}
 				if errors.Cause(err) == ErrInvalidAuthorization {
-					err = errors.Wrapf(err, "pull access denied, repository does not exist or may require authorization")
+					log.G(ctx).Info("pull access denied, repository does not exist or may require authorization, trying next host")
+					continue
```

Comment:
This `continue` is redundant as there is a `continue` further below on line 297.